### PR TITLE
Handle `-autolaunch YES` application argument

### PR DESCRIPTION
**Pull request by trunkmaster**
Map application icon window without focus switch for applications executed with argument `-autolaunch YES` in WindowMaker environment.

### CODE REVIEW

File: Source/x11/XGServerWindow.m [hunk 1]
```diff
@@ -2815,6 +2815,24 @@ - (void) orderwindow: (int)op : (int)otherWin : (int)winNum
...
 	{
+          NSUserDefaults *defaults = [NSUserDefaults standardUserDefaults];
+          if (op != NSWindowOut && window->map_state == IsUnmapped &&
+              [[defaults objectForKey: @"autolaunch"] isEqualToString:@"YES"])
+ 	    {
+              XEvent ev;
+
+              // Inform WM to ignore focus events
+              ev.xclient.type = ClientMessage;
+              ev.xclient.message_type = XInternAtom(dpy,"WM_IGNORE_FOCUS_EVENTS", False);
```
Comment (fredkiefer) [inline Source/x11/XGServerWindow.m]: In GNUstep back I did remove all the local calls to XInternAtom and get all the required atoms once at the start of the application, see _setupRootWindow. These atoms are defined in the file XGGeneric.h.
The only place my Google search found this specific atom "WM_IGNORE_FOCUS_EVENTS" was in WindowMaker source code. Is it only supported for this window manager? I also didn't find it in my X Windows reference book. Where does this come from?

Comment (trunkmaster) [reply Source/x11/XGServerWindow.m]: Yes it's WindowMaker only atom (defined here: https://repo.or.cz/wmaker-crm.git/blob/HEAD:/src/startup.c#l399).
Anyway this `if (generic.flags.useWindowMakerIcons == 1)` block (starts at line 2808) will be invoked only for WindowMaker. I suppose `useWindowMakerIcons` flag make sense only for WindowMaker.

**Comment by ivucica** [comment]
Would it make sense, then, to just expand "// Inform WM to ignore focus
events" to "// Inform WindowMaker to ignore focus events", making it even
clearer this is a WindowMaker-only atom?

>

**Comment by trunkmaster** [comment]
@ivucica No problem. But it make sense only for commit diff... In comment above this WindowMaker already mentioned. Anyway, I'll make change if code becomes more readable.

**Comment by trunkmaster** [comment]
@fredkiefer, new atom added.
@ivucica, comments edited.

**Pull request merged**